src/bpf/hidrecord.bpf.c

```diff
@@ -1,46 +1,14 @@
 // SPDX-License-Identifier: GPL-2.0-only
 /* Copyright (c) 2024 Benjamin Tissoires
  */
-#include "vmlinux.h"
-#include <bpf/bpf_tracing.h>
-
-char LICENSE[] SEC("license") = "GPL";
-
-struct {
-	__uint(type, BPF_MAP_TYPE_RINGBUF);
-	__uint(max_entries, 256 * 1024);
-} events SEC(".maps");
-
-struct hid_recorder_event {
-	__u8 length;
-	__u8 data[64];
-};
-
-extern __u8 *hid_bpf_get_data(struct hid_bpf_ctx *ctx,
-			      unsigned int offset,
-			      const size_t __sz) __ksym;
+#include "hidrecord.h"
 
 #define BPF_F_BEFORE (1U << 3)
 
-SEC("struct_ops/hid_device_event")
+SEC("?struct_ops/hid_device_event")
```

Comment:
It's a leftover from when I tried to have only one bpf program instead of two.
You can add a `?` in the SEC which tells libbpf to not load the program (autoload is `false`). However it wasn't working as expected for 2 reasons:
- you can not not autoload the struct_ops declaration (technically you can manually do that later in the code)
- it complained that the `syscall` to attach the tracing version of the filter was not having the proper kernel kfunc.
So I went the easy road and split the file in 2.

But the more I think of it, the more I think I must have done something wrong and might still be able to have a single `.bpf.c` file.